python/src/SeticorePy/viewer.py

```diff
@@ -13,12 +11,13 @@
 import os
 import pandas as pd
 
-SETICORE_DIR = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
-hit_capnp = capnp.load(SETICORE_DIR + "/hit.capnp")
-stamp_capnp = capnp.load(SETICORE_DIR + "/stamp.capnp")
+from importlib_resources import files as package_files
+
+hit_capnp = capnp.load(str(package_files('SeticorePy.schema').joinpath("hit.capnp")))
+stamp_capnp = capnp.load(str(package_files('SeticorePy.schema').joinpath("stamp.capnp")))
 
-def read_hits(filename):
-    with open(filename) as f:
+def read_hits(filepath):
```

Comment:
why "filepath"? I think "filename" is the right way to describe this entity.

Reply:
I've had to hold both the path to a file (`filepath = "/mnt/buf0/guppi_test_file.0000.raw"`) and the filename (`filename = "guppi_test_file.0000.raw"`) before so automatically I adjusted (oh, in my head there is also `filedir = "/mnt/buf0/"`, sort of all in line with the os.path.* functions).
It's your code base, I'll revert if you wish.

Reply:
So I sort of have two thoughts about this. One is that "path" and "name" don't actually refer to different things - I think the right terminology is, the "filename" and "pathname" refer to the same thing. "filename" just implies that it's a path to a file, rather than a path to a directory. There are two types of path/filename, the "absolute path" or "absolute filename" and the "relative path" or "relative filename". So when you say "path" that doesn't even imply that it's an absolute path, which I think is what you're getting at.

The second issue though is that the path here doesn't have to be absolute. `read_hits` works for either relative or absolute paths, and so do most of the functions that take filenames here. It's kind of the default to write code to take either sort of filename, that's ideal if you can do it imo. So it's also a misnomer to imply here that the pathname is an absolute one.

"filepath" doesn't mean anything afaict. If you mean that some path has to be the absolute path then the right terminology is absolute pathname or absolute filename. But I don't actually see anywhere in this code that absolute paths are required. In find_stamp_recipe you *could* use absolute paths but it doesn't look like the code actually does.

Reply:
I agree that "filepath" could infer the directory path of the file... but then what would an appropriate variable name be for something that is the file's full path? "file_uri"?

This is how I see it all:
- A file has a 'name' and sits in a directory. The file's name doesn't have any path/uri associated characters (the slashes that demarcate nested directories). The file's extension is a convention within its name, that the last '.' delimited part is the extension..
- A directory has a name and sits in a directory or at the root of a filesystem.
- A path is a URI encoded string that details a relative position of a directory or file (a resource) by detailing the traversal of directories to the resource. It can be relative in which case its first directory is omitted or is `.` or `..`, or it can be absolute, starting from the root of the filesystem in which case it starts with the directory delimiter. 

This is I believe how Python's built-in functions model it all too.